--- RtMidi.cpp
+++ RtMidi.cpp
@@ -2731,7 +2731,7 @@
   data->sysexBuffer.clear();
   data->sysexBuffer.resize( inputData_.bufferCount );
 
-  for ( int i=0; i < inputData_.bufferCount; ++i ) {
+  for ( unsigned int i=0; i < inputData_.bufferCount; ++i ) {
	 data->sysexBuffer[i] = (MIDIHDR*) new char[ sizeof(MIDIHDR) ];
	 data->sysexBuffer[i]->lpData = new char[ inputData_.bufferSize ];
	 data->sysexBuffer[i]->dwBufferLength = inputData_.bufferSize;
@@ -2785,7 +2785,7 @@
   }
 
   if ( data->sysexBuffer.size() ) {
-    for ( int i=0; i < data->sysexBuffer.size(); ++i ) {
+    for ( size_t i=0; i < data->sysexBuffer.size(); ++i ) {
	   int result = midiInUnprepareHeader(data->inHandle, data->sysexBuffer[i], sizeof(MIDIHDR));
	   delete [] data->sysexBuffer[i]->lpData;
	   delete [] data->sysexBuffer[i];